### PR TITLE
SDK-340: Add Spotless for code formatting and update documentation

### PR DESCRIPTION
**Pull request by mherman22**
## Description of what I changed

Add Spotless for code formatting and update documentation

- Integrate Spotless Maven plugin for automated code formatting
- Configure Spotless to use OpenMRS coding standards
- Add license header application to Java and XML files
- Update README.md with instructions on running Spotless
- Remove formatter-maven-plugin in favor of Spotless
- Ensure Spotless runs automatically during the build process
- Update pom.xml to include Spotless configuration

These changes improve code consistency and make it easier for
contributors to adhere to the project's coding standards.

## Issue I worked on

see https://openmrs.atlassian.net/browse/SDK-340

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean install` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute the above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: archetype-module-content/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -115,27 +115,42 @@
...
+                    <java>
+                        <licenseHeader>
+                            <file>${project.basedir}/license-header.txt</file>
+                        </licenseHeader>
+                        <eclipse>
+                            <file>${project.basedir}/OpenMRSFormatter.xml</file>
+                        </eclipse>
+                        <removeUnusedImports/>
+                        <importOrder>
+                            <order>javax,java,org,com,</order>
```
Comment (ibacher) [inline archetype-module-content/src/main/resources/archetype-resources/pom.xml:133]: I think this better matches our current sorting.

```suggestion
                            <order>javax,java</order>
```

File: archetype-module-content/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -115,27 +115,42 @@
...
-                    <removeUnused>true</removeUnused>
+                    <java>
+                        <licenseHeader>
+                            <file>${project.basedir}/license-header.txt</file>
+                        </licenseHeader>
+                        <eclipse>
+                            <file>${project.basedir}/OpenMRSFormatter.xml</file>
+                        </eclipse>
+                        <removeUnusedImports/>
+                        <importOrder>
```
Comment (ibacher) [inline archetype-module-content/src/main/resources/archetype-resources/pom.xml:132]: Don't we want `<removeUnusedImports />` in this block too?

File: archetype-module-refapp/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -124,70 +124,61 @@
...
 		</pluginManagement>
 	</build>
 
 	<repositories>
 		<repository>
 			<id>openmrs-repo</id>
-			<name>OpenMRS Nexus Repository</name>
-			<url>https://mavenrepo.openmrs.org/public</url>
+				<name>OpenMRS Nexus Repository</name>
+				<url>https://mavenrepo.openmrs.org/public</url>
```
Comment (ibacher) [inline archetype-module-refapp/src/main/resources/archetype-resources/pom.xml:181]: ????

**Review by ibacher — CHANGES_REQUESTED** [review]
Thanks @mherman22! Looks almost exactly right, but a couple of small tweaks.

File: archetype-module-content/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (mherman22) [reply archetype-module-content/src/main/resources/archetype-resources/pom.xml:132]: we actually don't , ref -> https://github.com/diffplug/spotless/blob/06168eb5ab9e932aee31a0ac2ac25fee67f45475/plugin-maven/src/main/java/com/diffplug/spotless/maven/java/ImportOrder.java#L29.

**Review by ibacher — APPROVED** [review]

**Pull request merged**